### PR TITLE
feat:upgrade gradle to 8.0.1

### PR DESCRIPTION
**Pull request by jakubuid**
Fix building release artefacts on Gradle 8+

### CODE REVIEW

File: jitpack.yml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 jdk:
-  - openjdk11
```
Comment (TalhaAli00) [inline jitpack.yml:2]: let's delete the jitpack file

File: samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/Web3WalletActivity.kt [hunk 2]
```diff
@@ -129,7 +128,7 @@ class Web3WalletActivity : ComponentActivity() {
                     }
                     is PushMessage -> {
                         val notificationBuilder = NotificationCompat.Builder(this, "Push")
-                            .setSmallIcon(R.drawable.ic_walletconnect_circle_blue)
+                            .setSmallIcon(com.walletconnect.sample_common.R.drawable.ic_walletconnect_circle_blue)
```
Comment (TalhaAli00) [inline samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/Web3WalletActivity.kt:131]: why did the import change?

File: samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/composable_routes/connection_details/ConnectionDetailsRoute.kt [hunk 3]
```diff
@@ -208,7 +220,7 @@ fun TopButtons(navController: NavController, isEmitVisible: Boolean, onEmit: ()
             .clickable(interactionSource = interactionSourceRow, indication = null) { navController.popBackStack() }
             .padding(5.dp)
         ) {
-            Icon(tint = color, imageVector = ImageVector.vectorResource(id = R.drawable.chevron_left), contentDescription = "Go back")
+            Icon(tint = color, imageVector = ImageVector.vectorResource(id = com.walletconnect.sample_common.R.drawable.chevron_left), contentDescription = "Go back")
```
Comment (TalhaAli00) [inline samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/composable_routes/connection_details/ConnectionDetailsRoute.kt:223]: fix import

File: samples/web3inbox/build.gradle.kts [hunk 4]
```diff
@@ -93,7 +94,10 @@ dependencies {
...
 
-    releaseImplementation(platform("com.walletconnect:android-bom:$BOM_VERSION"))
-    releaseImplementation("com.walletconnect:android-core")
-    releaseImplementation("com.walletconnect:web3inbox")
+    releaseImplementation(project(":androidCore:sdk"))
+    releaseImplementation(project(":web3:inbox"))
+
+//    releaseImplementation(platform("com.walletconnect:android-bom:$BOM_VERSION"))
+//    releaseImplementation("com.walletconnect:android-core")
+//    releaseImplementation("com.walletconnect:web3inbox")
```
Comment (TalhaAli00) [inline samples/web3inbox/build.gradle.kts:102]: let's revert this change

File: samples/web3wallet/build.gradle.kts [hunk 5]
```diff
@@ -97,8 +97,13 @@ dependencies {
...
-    releaseImplementation("com.walletconnect:web3inbox")
+    releaseImplementation(project(":androidCore:sdk"))
+    releaseImplementation(project(":web3:wallet"))
+    releaseImplementation(project(":web3:inbox"))
+    releaseImplementation(project(":push:sdk"))
+
+//    releaseImplementation(platform("com.walletconnect:android-bom:$BOM_VERSION"))
+//    releaseImplementation("com.walletconnect:android-core")
+//    releaseImplementation("com.walletconnect:web3wallet")
+//    releaseImplementation("com.walletconnect:web3inbox")
```
Comment (TalhaAli00) [inline samples/web3wallet/build.gradle.kts:108]: revert

File: sign/sdk/proguard-rules.pro
Comment (TalhaAli00) [inline sign/sdk/proguard-rules.pro:1]: let's move these rules into the wc rules file in consumer-rules. Then we can delete this file

File: sign/wallet/src/main/kotlin/com/walletconnect/wallet/WalletFirebaseMessagingService.kt [hunk 6]
```diff
@@ -39,7 +39,7 @@ class WalletFirebaseMessagingService: PushMessageService() {
         val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
         val notificationBuilder = NotificationCompat.Builder(this, channelId)
             .setContentTitle(message.title)
-            .setSmallIcon(R.drawable.ic_walletconnect_circle_blue)
+            .setSmallIcon(com.walletconnect.sample_common.R.drawable.ic_walletconnect_circle_blue)
```
Comment (TalhaAli00) [inline sign/wallet/src/main/kotlin/com/walletconnect/wallet/WalletFirebaseMessagingService.kt:42]: fix import

File: sign/wallet/src/main/kotlin/com/walletconnect/wallet/ui/host/WalletSampleActivity.kt [hunk 7]
```diff
@@ -74,7 +78,7 @@ class WalletSampleActivity : AppCompatActivity() {
                     }
                     is SampleWalletEvents.PushMessage -> {
                         val notificationBuilder = NotificationCompat.Builder(this, "Push")
-                            .setSmallIcon(R.drawable.ic_walletconnect_circle_blue)
+                            .setSmallIcon(com.walletconnect.sample_common.R.drawable.ic_walletconnect_circle_blue)
```
Comment (TalhaAli00) [inline sign/wallet/src/main/kotlin/com/walletconnect/wallet/ui/host/WalletSampleActivity.kt:81]: fix import

**Comment by TalhaAli00** [comment]
For all of the `com.walletconnect.sample_common.R` imports, can you create an alias and change the import for each of the instances?

**Review by TalhaAli00 — APPROVED** [review]

**Review by Elyniss — APPROVED** [review]
